pandas/io/common.py

```diff
@@ -328,6 +372,7 @@ def _get_handle(path_or_buf, mode, encoding=None, compression=None,
     path_or_buf = _stringify_path(path_or_buf)
     is_path = isinstance(path_or_buf, str)
 
+    compression, compression_args = _get_compression_method(compression)
```

Comment:
Related to comment above, I think it would make sense to do the isinstance check before calling this method, unless you think there is a reason why we would ever need this method to do anything with a `str` passed through as an argument